src/latest-release.ts

```diff
@@ -0,0 +1,25 @@
+import * as nodegit from "nodegit";
+import * as semver from "semver";
+
+
+export async function checkoutLatestRelease(
+    repo: nodegit.Repository,
+    dir?: string)
+        : Promise<nodegit.Repository> {
+  const tags = await nodegit.Tag.list(repo);
+  let latestRelease: string;
+  if (tags && tags.length > 0) {
+    try {
+      latestRelease = tags.sort(semver.rcompare)[0];
+    } catch (err) {
+      // We couldn't pick a release, so we'll just bow out here.
+      return repo;
```

Comment:
semver.rcompare can throw, sadly. I've added a comment and a TODO to resolve this.